docs/docs/aztec/protocol/errors.md

```diff
@@ -0,0 +1,169 @@
+
+#### 3002 - PUBLIC_KERNEL__PRIVATE_FUNCTION_NOT_ALLOWED
+Calling a private noir function in a public kernel is not allowed. 
+
+#### 3005 - PUBLIC_KERNEL__NON_EMPTY_PRIVATE_CALL_STACK
+Public functions are executed after all the private functions are (see [private-public execution](../how-it-works/private-public-execution.mdx)). As such, private call stack must be empty when executing in the public kernel.
+
+#### 3011 - PUBLIC_KERNEL__CALCULATED_PRIVATE_CALL_HASH_AND_PROVIDED_PRIVATE_CALL_HASH_MISMATCH
+When the hash stored at the top most of the call stack is different to the call stack item expected by the public kernel's inputs.
+
+#### 3012 - PUBLIC_KERNEL__PUBLIC_CALL_STACK_MISMATCH
+Similar to above, except here we actually have the preimages to the call stack and hash to ensure they match.
+
+#### 3013 - PUBLIC_KERNEL__CONTRACT_DEPLOYMENT_NOT_ALLOWED
+Public kernel doesn't allow contract deployments
+
+#### 3014 - PUBLIC_KERNEL__CONSTRUCTOR_NOT_ALLOWED
+Aztec doesn't support public constructors.
+
+#### 3015 - PUBLIC_KERNEL__CONTRACT_ADDRESS_INVALID
+Calling `address(0x0)` publicly is not permitted.
+
+#### 3016 - PUBLIC_KERNEL__FUNCTION_SIGNATURE_INVALID
+Cannot call a contract with no function (i.e. function signature of 0) publicly. 
+
+#### 3022 - PUBLIC_KERNEL__PUBLIC_CALL_STACK_CONTRACT_STORAGE_UPDATES_PROHIBITED_FOR_STATIC_CALL
+For static calls, no contract storage change requests are allowed. 
+
+#### 3024 - PUBLIC_KERNEL__CALL_CONTEXT_CONTRACT_STORAGE_UPDATE_REQUESTS_PROHIBITED_FOR_STATIC_CALL
+Same as [3022](#3022---public_kernel__public_call_stack_contract_storage_updates_prohibited_for_static_call), no contract changes are allowed for static calls.
+
+#### 3026 - PUBLIC_KERNEL__NEW_COMMITMENTS_PROHIBITED_IN_STATIC_CALL
+For static calls, no new commitments or nullifiers can be added to the state.
+
+#### 3027 - PUBLIC_KERNEL__NEW_NULLIFIERS_PROHIBITED_IN_STATIC_CALL
+For static calls, no new commitments or nullifiers can be added to the state.
+
+### Rollup circuit errors
+These are errors that occur when kernel proofs (transaction proofs) are sent to the rollup circuits to create an L2 block. See [rollup circuits](./circuits/rollup.md) for more information.
+
+#### 4007 - BASE__INVALID_CHAIN_ID
+The L1 chain ID you used in your proof generation (for your private transaction) is different to what the rollup circuits expected. Double check against the global variables passed to noir and the config set in [Aztec's rollup contract](https://github.com/AztecProtocol/aztec-packages/blob/master/l1-contracts/src/core/Rollup.sol) which are [read in by sequencer](https://github.com/AztecProtocol/aztec3-packages/blob/master/yarn-project/sequencer-client/src/global_variable_builder/global_builder.ts#L32) and subsequently passed in as inputs to the base rollup. When the sequencer submits the block to the rollup contracts, this is again sanity checked so ensure this is the same everywhere. 
```

Comment:
Is this chain ID definitely L1 chain ID? I always assumed we would track an L2 chain ID

Reply:
yea l2 chain id is called "version" for now